macros/edr/tests/test_configuration/get_time_bucket.sql

```diff
@@ -0,0 +1,67 @@
+{% macro get_daily_time_bucket() %}
+  {% do return({"period": "day", "count": 1}) %}
+{% endmacro %}
+
+{% macro get_default_time_bucket() %}
+  {% do return(elementary.get_daily_time_bucket()) %}
+{% endmacro %}
+
+{% macro get_time_bucket(time_bucket, model_graph_node) %}
+    {%- set configured_time_bucket = elementary.get_test_argument('time_bucket', time_bucket, model_graph_node) %}
+    {%- do elementary.validate_time_bucket(configured_time_bucket) %}
+    {%- set default_time_bucket = elementary.get_default_time_bucket() %}
+
+    {%- if not configured_time_bucket %}
+        {{ return(default_time_bucket) }}
+    {%- elif configured_time_bucket and not configured_time_bucket.period %}
```

Comment:
There's no need to re-check if `configured_time_bucket` because you test that above.
Also, you can do this if you want:
```j2
{% set time_bucket = default_time_bucket.copy() %}
{% do time_bucket.update(configured_time_bucket) %}
```